packages/shared.net/src/createClientContext.ts

```diff
@@ -39,22 +38,26 @@ export function createClientContext<
     throw new Error("stack cannot be empty");
   }
 
-  const wireUserAgent = `${ontology.metadata.userAgent} ${userAgent}`;
-
   const retryingFetchWithAuthOrThrow = createFetchHeaderMutator(
     createRetryingFetch(createFetchOrThrow(fetchFn)),
     async (headers) => {
       const token = await tokenProvider();
       headers.set("Authorization", `Bearer ${token}`);
+
       headers.set(
         "Fetch-User-Agent",
-        wireUserAgent,
+        [
+          headers.get("Fetch-User-Agent"),
+          ontology.metadata.userAgent,
+          userAgent,
+        ].filter(x => x && x?.length > 0).join(" "),
```

Comment:
This keeps the headers in the same format as they previously were, except we can delay assigning now.